aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/TransferManager.java

```diff
@@ -1178,17 +1192,22 @@ public MultipleFileDownload downloadDirectory(String bucketName, String keyPrefi
      *            recursively.
      * @param destinationDirectory
      *            The directory to place downloaded files. Subdirectories will
-     *            be created as necessary.
+     *            be created as necessary. The created directory structure
+     *            will match that of the entire keys. See {@code stripKeyPrefix}
+     *            to change this behavior.
      * @param resumeOnRetry
      *            If set to true, upon an immediate retry of a failed object
      *            download, the <code>TransferManager</code> will resume the
      *            download from the current end of the file on disk.
      * @param filter
      *           If set, applies the filter to determine which keys to include
      *           in the download request. (default is include all).
+     * @param stripKeyPrefix
+     *           If set, strips the {@code keyPrefix} from the directory structure
+     *           of the downloaded files.
      */
     public MultipleFileDownload downloadDirectory(String bucketName, String keyPrefix, File destinationDirectory,
-                                                  boolean resumeOnRetry, KeyFilter filter) {
+                                                  boolean resumeOnRetry, KeyFilter filter, boolean stripKeyPrefix) {
```

Comment:
You never use this flag to strip the key prefix from the object key 😕

Reply:
Oops that's embarrassing! That's what I get for re-doing my changes in a more pull request worthy way...

Thanks, I pushed the actual functionality.